okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/OkHttpClients.java

```diff
@@ -152,24 +153,21 @@ public static OkHttpClient create(
             Class<?> serviceClass) {
         boolean reshuffle =
                 !config.nodeSelectionStrategy().equals(NodeSelectionStrategy.PIN_UNTIL_ERROR_WITHOUT_RESHUFFLE);
-        return createInternal(
-                client,
-                config,
-                userAgent,
-                hostEventsSink,
-                serviceClass,
-                RANDOMIZE,
-                reshuffle,
-                () -> new ExponentialBackoff(config.maxNumRetries(), config.backoffSlotSize()));
+        ClientConfiguration config1 = ClientConfiguration.builder()
+                .from(config)
+                .userAgent(userAgent)
+                .build();
+
+        return createInternal(client, config1, hostEventsSink, serviceClass, RANDOMIZE, reshuffle, () ->
+                new ExponentialBackoff(config1.maxNumRetries(), config1.backoffSlotSize()));
     }
 
     @VisibleForTesting
     static RemotingOkHttpClient withStableUris(
-            ClientConfiguration config, UserAgent userAgent, HostEventsSink hostEventsSink, Class<?> serviceClass) {
+            ClientConfiguration config, HostEventsSink hostEventsSink, Class<?> serviceClass) {
```

Comment:
thoughts on also including the HostEventsSink thingy in ClientConfiguration?

Reply:
I kinda like the idea. It's a good place to add new optional config especially as we can provide defaults too, and reduce the method argument inflation that's been going on here.